### DISCUSSION TITLE
Ramping up on BJS

### DISCUSSION THREAD

**Discussion by jelster** [jelster/space-truckers#21] [Playground Snippets]
![image](https://user-images.githubusercontent.com/1508500/106286081-dfde1280-620a-11eb-99fc-2b53bcaa45a3.png)

Loading / Background snippet built for Ramping up on BabylonJS chapter material. 
[https://playground.babylonjs.com/#0UYAPE#42](https://playground.babylonjs.com/#0UYAPE#42)

ed: updated the PG snippet revision to the current latest

**Comment by geonya** [comment]
https://playground.babylonjs.com/#7FU8ZA#24

I love this chapter.
The planets orbits are so cool!
thanks for your teaching!

**Reply by jelster** [reply]
I like the color palette changes you made - gives it a subtly smoother feel!

**Reply by geonya** [reply]
Thanks!! 
I added TransformNode to fix trail twisting bug.
https://playground.babylonjs.com/#7FU8ZA#26
I'm not sure It's right ... 😅

**Reply by jelster** [reply]
> I'm not sure It's right ... 😅

You're exactly right using a TransformNode to eliminate the twisty trails! That's exactly what I had in mind with that particular Extended Topic 😄 

One small tweak you can make is to use the loop index to give each TransformNode a unique name instead of them all being named `root` (that makes me think "I am Groot!" lol). It's not a big deal if they do have the same name, it will just be more difficult to distinguish them in the Inspector as well as complicating slightly the use of things like `scene.getMeshByName`. 

Nice work! 👍

**Reply by geonya** [reply]
I got it ...!
I thought that TransformNode should be only one element. 😅
I'm going to use a combination of planet names and loop index!
Thanks for this wonderful course ! 
I'll do my best to finish this course 🔥

**Reply by e4779** [reply]
Hi! I'm looked at your example.

Using TranformNode not fixed trail twisting. To be more accurate, trail is stop twisting, because planets are just stop rotating. This is not notable, because at your example planets are smooth, but it noticeable at Josh example ([here is tweak,](https://playground.babylonjs.com/#0UYAPE#115) where I add TransformNode)

**Reply by e4779** [reply]
It seems that I was [able](https://playground.babylonjs.com/#GVK38B#1) to use the TransformNode in such a way that both the planet rotates and the trail does not twist.

I specified the created transformNode as the parent for each planet (line 94:95). And use it for orbital animation and as generator for TrailMesh.

Maybe, there is a better way to define TransformNode?

**Reply by jelster** [reply]
That indeed is a good way to go about it @e41q - making the trail mesh a sibling of the planet w/ the TransformNode and targeting the animation accordingly prevents the twisting since the trail doesn't participate in the planet's rotation transformation. 

though the spinAnimation was commented out, [restoring](https://playground.babylonjs.com/#GVK38B#2) it does indeed show the planets rotating without twisting up the TrailMesh!

**Reply by e4779** [reply]
My bad, forget to uncomment before save snippet 😉

**Comment by hugeen** [comment]
Here is mine https://playground.babylonjs.com/#XJX7EB#18

**Reply by jelster** [reply]
I like the star's blue-hot color and how it contrasts with the rest of the trail colors!
